### PR TITLE
Working version of amp control

### PR DESCRIPTION
**Pull request by oscarbodell**

### CODE REVIEW

File: client/assets/js/main.js [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+}
+
+function forceReconnect() {
+    nextConnectionAttempt = 0
+}
+
+function setupPage() {
+    let newContent = `<table class="w3-table-all w3-card">`
+    for (let i = 0; i < 7; i++) {
+        newContent += `<tr class=""><td>`
```
Comment (danba340) [inline client/assets/js/main.js:81]: `
You can have 
multiple
lines in backtick strings
`

File: client/assets/js/main.js [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            jsn["value"] = $(`#zone_${id}_${type}`).val()
+            jsn["type"] = type
+            sendCommand(id, jsn)
+        }, SEND_THROTTLE_MS)
+        nextInputSendAllowed = t + SEND_THROTTLE_MS;
+    }
+}
+
+function setState(stateAsJson) {
+    stateAsJson.forEach(function (state) {
```
Comment (danba340) [inline client/assets/js/main.js:194]: Like :)

File: client/assets/js/main.js [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+function setStatesFromJson(channel, key) {
+    setObjectState(channel, "volume", key)
+    setObjectState(channel, "source", key)
+    setObjectState(channel, "treble", key)
+    setObjectState(channel, "bass", key)
+    setObjectState(channel, "balance", key)
+}
+
+function setObjectState(channel, key, json) {
+    const object = $(`#zone_${channel}_${key}`)
```
Comment (danba340) [inline client/assets/js/main.js:221]: Why is this variable called object?

Comment (danba340) [reply client/assets/js/main.js:221]: Ah ok i get why

File: client/assets/js/main.js [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+        if (connectionAttempts == 0) {
+            showConnectingScreen()
+        }
+        attemptReconnect()
+    }
+}
+
+function attemptReconnect() {
+    setTimeout(() => {
+        const nextReconnectionTime = Math.round((nextConnectionAttempt - now()) / 1000)
```
Comment (danba340) [inline client/assets/js/main.js:42]: Is now() from a library?

File: client/assets/js/main.js [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+        }
+    }
+
+    webSocket.onopen = function () {
+        connectionAttempts = 0
+        setupPage()
+    }
+
+    webSocket.onclose = function () {
+        if (connectionAttempts == 0) {
```
Comment (danba340) [inline client/assets/js/main.js]: Always use 3 ===

File: client/assets/js/main.js (hunk 4 above)
Comment (oscarbodell) [reply client/assets/js/main.js:42]: It's defined in the very end of this file.
function now() {
    return (new Date).getTime();
}

**Pull request merged**